File: Server/DataService/DataService/Models/Enums.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataService.Models
{
    //For agency view
    public enum RequestStatusEnum
    {
        [Display(Name = "Chờ nhân viên xử lý")]
        Pending = 1,
        [Display(Name = "Đang xử lý")]
        Processing = 2,
        [Display(Name = "Hoàn thành")]
        Done = 3,
        [Display(Name = "Hủy bỏ")]
        Cancel = 4,
        [Display(Name = "Tạo mới")]
        New = 5,
        [Display(Name = "Chờ hủy")]
        WaitingCancel = 6,
        [Display(Name = "Chờ xác nhận hoàn thành")]
        WaitingDone = 7
    }

    //For ITSupporter View
    public enum RequestDoingStatus
    {
        [Display(Name = "Đã nhận")]
        Received = 1,
        [Display(Name = "Đang di chuyển")]
        Moving = 2,
        [Display(Name = "Đang sửa chữa")]
        Fixing = 3,
        [Display(Name = "Đợi linh kiện")]
        Holding = 4,
        [Display(Name = "Hoàn thành")]
        Done = 5
    }

    public enum TicketStatusEnum
    {
        [Display(Name = "Đang chờ xử lý")]
        Await = 1,
        [Display(Name = "Đang xử lý")]
        In_Process = 2,
        [Display(Name = "Hoàn thành")]
        Done = 3,
        [Display(Name = "Hủy bỏ")]
        Cancel = 4
    }

    public enum RequestTaskEnum
    {       
        [Display(Name = "Đang xử lý")]
        In_Process = 1,
        [Display(Name = "Hoàn thành")]
        Done = 2,
    }

    public enum Gender
    {
        [Display(Name = "Nam")]
        Male = 1,
        [Display(Name = "Nữ")]
        Female = 2,
        [Display(Name = "Khác")]
        Others = 3
    }

    public enum RequestPriorityEnum
    {
        [Display(Name = "Cao")]
        Urgent = 1,
        [Display(Name = "Trung bình")]
        Normal = 2,
        [Display(Name = "Thấp")]
        Non_Urgent = 3,
    }

}

File: Mobile/AGENCY_App/app/src/main/java/com/odts/utils/Enums.java
package com.odts.utils;

public class Enums {
    //For agency view
    public enum RequestStatusEnum {
        Pending("Chờ Xử lý", 1),
        Processing("Đang xử lý", 2),
        Done("Hoàn thành", 3),
        Cancel("Hủy bỏ", 4),
        New("Tạo mới", 5),
        WaitingCancel ("Chờ hủy", 6);

        private String stringValue;
        private int intValue;

        private RequestStatusEnum(String toString, int value) {
            stringValue = toString;
            intValue = value;
        }

        public String getStringValue() {
            return stringValue;
        }

        public int getIntValue() {
            return intValue;
        }

        @Override
        public String toString() {
            return stringValue;
        }
    }

    //For ITSupporter View
    enum TicketStatusEnum {
        Await("Đang chờ xử lý", 1),
        In_Process("Đang xử lý", 2),
        Done("Hoàn Thành", 3),
        Cancel("Hủy bỏ", 4);

        private String stringValue;
        private int intValue;

        private TicketStatusEnum(String toString, int value) {
            stringValue = toString;
            intValue = value;
        }

        public String getStringValue() {
            return stringValue;
        }

        public int getIntValue() {
            return intValue;
        }

        @Override
        public String toString() {
            return stringValue;
        }
    }

    //For Ticket Task
    enum TicketTaskEnum {
        New("Mới", 1),
        In_Process("Đang xử lý", 2),
        Done("Hoàn thành", 3),
        Cancel("Hủy bỏ", 4);

        private String stringValue;
        private int intValue;

        private TicketTaskEnum(String toString, int value) {
            stringValue = toString;
            intValue = value;
        }

        public String getStringValue() {
            return stringValue;
        }

        public int getIntValue() {
            return intValue;
        }

        @Override
        public String toString() {
            return stringValue;
        }
    }

    enum Gender {
        Male("Nam", 1),
        Female("Nữ", 2),
        Others("Khác", 3);

        private String stringValue;
        private int intValue;

        private Gender(String toString, int value) {
            stringValue = toString;
            intValue = value;
        }

        public String getStringValue() {
            return stringValue;
        }

        public int getIntValue() {
            return intValue;
        }

        @Override
        public String toString() {
            return stringValue;
        }
    }
}



